pkgs/applications/version-management/github-desktop/default.nix

```diff
@@ -0,0 +1,100 @@
+{ darkMode ? false, stdenv, fetchurl, dpkg, makeWrapper
+, alsaLib, atk, cairo, cups, curl, dbus, expat, fontconfig, freetype, glib
+, gnome2, gtk3, gdk_pixbuf, libnotify, libxcb, nspr, nss, pango
+, systemd, xorg }:
+
+let
+
+  version = "1.5.1";
+
+  rpath = stdenv.lib.makeLibraryPath [
+    alsaLib
+    atk
+    cairo
+    cups
+    curl
+    dbus
+    expat
+    fontconfig
+    freetype
+    glib
+    gnome2.GConf
+    gdk_pixbuf
+    gtk3
+    pango
+    libnotify
+    libxcb
+    nspr
+    nss
+    stdenv.cc.cc
+    systemd
+
+    xorg.libxkbfile
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXtst
+    xorg.libXScrnSaver
+  ] + ":${stdenv.cc.cc.lib}/lib64";
+
+  src =
+    if stdenv.hostPlatform.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://github.com/shiftkey/desktop/releases/download/release-1.5.1-linux2/GitHubDesktop-linux-${version}-linux2.deb";
+        sha256 = "0nbgv920l9l49046yv8nal990p6nxgsh6fw18bxpflg9h6qi4afq";
+      }
+    else
+      throw "Slack is not supported on ${stdenv.hostPlatform.system}";
```

Comment:
This still writes "Slack".

On the other hand, platforms are already listed in `meta.platforms`,  so you don't need to throw here?

Does upstream provide multiple per-arch builds? If it does, I'd propose acessing from an attrset, like in https://github.com/NixOS/nixpkgs/blob/2453b5ea7999218f7af3ea8f0895895b70b21d1e/pkgs/applications/networking/resilio-sync/default.nix#L14-L19, otherwise, just point to the one `x86_64-linux` source unconditionally.